bundler/lib/dependabot/bundler/update_checker/latest_version_finder/dependency_source.rb

```diff
@@ -98,46 +103,53 @@ def dependency_rubygems_uri
 
           def private_registry_versions
```

Comment:
Not pictured in the diff, there is a `rubygems_versions` which is the one method that doesn't call out to the native helper as we currently just use Excon to talk to it directly instead of using the `Bundler::Source#fetchers`.

I've left that as-is since moving it to the native helper doesn't give us any gain and leads to a huge test re-write where a lot of webstubbing needs to be replaced/moved. I figured it was better to minimise the diff for this pass.

More generally, I think it might be worth just using bundler's own fetchers to get information even from rubygems and some of this code might lean out, but that seems orthogonal to getting it working.